src/main/kotlin/net/atos/zac/app/informatieobjecten/EnkelvoudigInformatieObjectUpdateService.kt

```diff
@@ -44,15 +57,33 @@ class EnkelvoudigInformatieObjectUpdateService @Inject constructor(
             )
         }
 
-    fun ondertekenEnkelvoudigInformatieObject(uuid: UUID) {
+    fun ondertekenEnkelvoudigInformatieObject(informationObjectUUID: UUID) {
         EnkelvoudigInformatieObjectWithLockData().apply {
             ondertekening = Ondertekening().apply {
                 soort = Ondertekening.SoortEnum.DIGITAAL
                 datum = LocalDate.now()
             }
             status = EnkelvoudigInformatieObjectWithLockData.StatusEnum.DEFINITIEF
-            updateEnkelvoudigInformatieObjectWithLockData(uuid, this, ONDERTEKENEN_TOELICHTING)
+            updateEnkelvoudigInformatieObjectWithLockData(informationObjectUUID, this, ONDERTEKENEN_TOELICHTING)
+            CoroutineScope(Dispatchers.IO).launch { sendChangeEvent(informationObjectUUID) }
+        }
+    }
+
+    private suspend fun sendChangeEvent(informationObjectUUID: UUID) {
```

Comment:
maybe this can be refactored into the eventingService or a helper utility? Then the logic for repeat logic can be implemented once and used in both places, just need to pass the verification predicate...

Reply:
Thanks @bas-info-nl ! Much simpler this way